src/source/tcp.rs

```diff
@@ -134,44 +139,54 @@ where
             let mut events = mio::Events::with_capacity(1024);
             match poll.poll(&mut events, None) {
                 Err(e) => panic!(format!("Failed during poll {:?}", e)),
-                Ok(_num_events) => {
-                    for event in events {
-                        match event.token() {
-                            constants::SYSTEM => {
-                                for handler in self.handlers {
-                                    handler.shutdown();
-                                }
-
-                                util::send(&mut chans, metric::Event::Shutdown);
-                                return;
+                Ok(_num_events) => for event in events {
+                    match event.token() {
+                        constants::SYSTEM => {
+                            for handler in self.handlers {
+                                handler.shutdown();
                             }
-                            listener_token => {
-                                self.spawn_stream_handlers(
-                                    chans.clone(), // TODO: do not clone, make an Arc
-                                    tags,
-                                    listener_token,
-                                );
+
+                            util::send(&mut chans, metric::Event::Shutdown);
+                            return;
+                        }
+                        listener_token => {
+                            let listener = &(self.listeners[listener_token]);
+                            match self.spawn_stream_handlers(
+                                chans.clone(),
+                                tags,
+                                listener,
+                            ) {
+                                Ok(new_handlers) => {
+                                    self.handlers.extend(new_handlers);
+                                }
+                                Err(e) => {
+                                    error!("Failed to spawn stream handlers! {:?}", e);
```

Comment:
Under what conditions would we fail to spawn a stream handler? Far-side is partitioned away, too many files open. Anything else? 

Trying ultimately to figure out if this is a full-blown error logging condition – which people will ask about – or if it could be somewhat lower.

Reply:
All the cases when we fail to spawn a thread.  Maybe we are out of memory, fds, to name a couple.

Reply:
`ConnectionAborted` is an error kind that should be ignored actually. That means a connection arrived in the listen queue but was subsequently closed prior to accept being called.